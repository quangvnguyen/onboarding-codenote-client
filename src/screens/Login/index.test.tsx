import React from 'react';
import { mount, shallow } from 'enzyme';
import { Login } from './index';

describe('Login component', () => {

  it('renders without crashing', () => {
    const wrapper = shallow(
      <Login />
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('renders with form validated', () => {
    const wrapper = mount(
      <Login />
    );
    wrapper.setState({
      email: 'email',
      password: 'password',
    });
    expect(wrapper).toMatchSnapshot();
  });

  it('should handleChange function in email fields called', () => {
    const event = {target: {id: 'email', value: 'email'}};
    const wrapper = mount(
      <Login />
    );

    wrapper.find('#email').simulate('change', event);
    expect(wrapper.state().email).toEqual('email');
  });


  it('should handleChange function in password fields called', () => {
    const event = {target: {id: 'password', value: 'password'}};

    const wrapper = mount(
      <Login />
    );
    wrapper.find('#password').simulate('change', event);
    expect(wrapper.state().password).toEqual('password');
  });

  it('should handleSubmit function called', () => {
    const props = {
      userHasAuthenticated: jest.fn(),
    };

    const wrapper = mount(
      <Login {...props} />
    );

    const component = wrapper.find('form');
    component.simulate('submit');

    expect(props.userHasAuthenticated).toHaveBeenCalled();
  });
})