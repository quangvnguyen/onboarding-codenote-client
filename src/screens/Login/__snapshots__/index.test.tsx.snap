// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login component renders with form validated 1`] = `
<Login>
  <div
    className="Login"
  >
    <form
      onSubmit={[Function]}
    >
      <FormGroup
        bsClass="form-group"
        bsSize="large"
        controlId="email"
      >
        <div
          className="form-group form-group-lg"
        >
          <ControlLabel
            bsClass="control-label"
            srOnly={false}
          >
            <label
              className="control-label"
              htmlFor="email"
            >
              Email
            </label>
          </ControlLabel>
          <FormControl
            autoFocus={true}
            bsClass="form-control"
            componentClass="input"
            onChange={[Function]}
            type="email"
            value="email"
          >
            <input
              autoFocus={true}
              className="form-control"
              id="email"
              onChange={[Function]}
              type="email"
              value="email"
            />
          </FormControl>
        </div>
      </FormGroup>
      <FormGroup
        bsClass="form-group"
        bsSize="large"
        controlId="password"
      >
        <div
          className="form-group form-group-lg"
        >
          <ControlLabel
            bsClass="control-label"
            srOnly={false}
          >
            <label
              className="control-label"
              htmlFor="password"
            >
              Password
            </label>
          </ControlLabel>
          <FormControl
            bsClass="form-control"
            componentClass="input"
            onChange={[Function]}
            type="password"
            value="password"
          >
            <input
              className="form-control"
              id="password"
              onChange={[Function]}
              type="password"
              value="password"
            />
          </FormControl>
        </div>
      </FormGroup>
      <LoaderButton
        block={true}
        bsSize="large"
        disabled={false}
        isLoading={false}
        loadingText="Logging in ..."
        text="Login"
        type="submit"
      >
        <Button
          active={false}
          block={true}
          bsClass="btn"
          bsSize="large"
          bsStyle="default"
          className="LoaderButton "
          disabled={false}
          type="submit"
        >
          <button
            className="LoaderButton  btn btn-lg btn-default btn-block"
            disabled={false}
            type="submit"
          >
            Login
          </button>
        </Button>
      </LoaderButton>
    </form>
  </div>
</Login>
`;

exports[`Login component renders without crashing 1`] = `
<div
  className="Login"
>
  <form
    onSubmit={[Function]}
  >
    <FormGroup
      bsClass="form-group"
      bsSize="large"
      controlId="email"
    >
      <ControlLabel
        bsClass="control-label"
        srOnly={false}
      >
        Email
      </ControlLabel>
      <FormControl
        autoFocus={true}
        bsClass="form-control"
        componentClass="input"
        onChange={[Function]}
        type="email"
        value=""
      />
    </FormGroup>
    <FormGroup
      bsClass="form-group"
      bsSize="large"
      controlId="password"
    >
      <ControlLabel
        bsClass="control-label"
        srOnly={false}
      >
        Password
      </ControlLabel>
      <FormControl
        bsClass="form-control"
        componentClass="input"
        onChange={[Function]}
        type="password"
        value=""
      />
    </FormGroup>
    <LoaderButton
      block={true}
      bsSize="large"
      disabled={true}
      isLoading={false}
      loadingText="Logging in ..."
      text="Login"
      type="submit"
    />
  </form>
</div>
`;
